nilearn/input_data/tests/test_nifti_region.py

```diff
@@ -199,6 +200,22 @@ def test_nifti_labels_masker_2():
                  (masker.labels_img_.shape[:3] + (length,)))
 
 
+    # Test with data and atlas of different shape: the atlas should be
+    # resampled to the data
+    shape22 = (5, 5, 6)
+    affine2 = 2 * np.eye(4)
+    affine2[-1, -1] = 1
+
+    fmri22_img, _ = generate_random_img(shape22, affine=affine2,
+                                                 length=length)
+    masker = NiftiLabelsMasker(labels33_img, mask_img=mask22_img)
+
+    masker.fit_transform(fmri22_img)
+    np.testing.assert_array_equal(
+        masker._resampled_labels_img_.get_affine(),
+        affine2)
+
+
```

Comment:
Do we want a test that uses an explicit resampling_target="data" to check that it is the same as not specifying resampling_target?